### DISCUSSION TITLE
Fixing parameters

### DISCUSSION THREAD

**Discussion by anne8rete** [LAPKB/Pmetrics#80] [Q&A] [answered]
Hi, 

I am trying to fix a parameter in the Version 2 Pmetrics, how to do this in the code/txt file? The previous use of "!" doesn't seem to work, also putting into code ratio=0.3 doesn't work.

Example (wanting to fix the ratio): 

mod3_met <- PM_model$new(list(
  pri = list(
    V = range(0, 200),
    K12 = range(0,10),
    K21 = range(0,10),
    Vmax = range(0,100),
    Km = range(0,80),
    K34 = range(0,10),
    K43 = range(0,10),
    SCLM = range(0,100),
    ratio = range(0.3,0.4)
  ),
  sec = list("V2 = ratio*V"),
  dif = list(
    xp1="RATEIV(1) - (K12 + Vmax/(Km*V+X(1))) *X(1) + K21*X(2)",
    xp2="K12*X(1) - K21*X(2)",
    xp3="(Vmax/(Km*V+X(1)))*X(1) - K43*X(4) - K34*X(3) - (SCLM/V2)*X(3)",
    xp4="K34*X(3) - K43*X(4)"
    ),
  out = list(
    Y1 = list(
      value = "X(1)/V",
      err = list(
        model = proportional(5),
        assay = c(0.01, 0.1, 0, 0)
      )
    ),
    Y2 = list(
      value = "X(3)/V2",
      err = list(
        model = proportional(5),
        assay = c(0.01, 0.1, 0, 0)
    )
  )
  )
))

Thanks, 
Anne

**Comment by mnneely** [comment, marked as answer]
Use ratio = fixed(3) if you want it to be estimated, but the same for all subjects, starting at a value of 3. Use ratio = fixed(3, constant = T) if you don’t want to estimate it and set the value to 3 for all subjects. Sorry that we are still catching up on documentation!

On Jul 13, 2022, at 3:42 AM, anne8rete ***@***.***> wrote:

﻿

Hi,

I am trying to fix a parameter in the Version 2 Pmetrics, how to do this in the code/txt file? The previous use of "!" doesn't seem to work, also putting into code ratio=0.3 doesn't work.

Example (wanting to fix the ratio):

mod3_met <- PM_model$new(list(
pri = list(
V = range(0, 200),
K12 = range(0,10),
K21 = range(0,10),
Vmax = range(0,100),
Km = range(0,80),
K34 = range(0,10),
K43 = range(0,10),
SCLM = range(0,100),
ratio = range(0.3,0.4)
),
sec = list("V2 = ratioV"),
dif = list(
xp1="RATEIV(1) - (K12 + Vmax/(KmV+X(1))) X(1) + K21X(2)",
xp2="K12X(1) - K21X(2)",
xp3="(Vmax/(KmV+X(1)))X(1) - K43X(4) - K34X(3) - (SCLM/V2)X(3)",
xp4="K34X(3) - K43*X(4)"
),
out = list(
Y1 = list(
value = "X(1)/V",
err = list(
model = proportional(5),
assay = c(0.01, 0.1, 0, 0)
)
),
Y2 = list(
value = "X(3)/V2",
err = list(
model = proportional(5),
assay = c(0.01, 0.1, 0, 0)
)
)
)
))

Thanks,
Anne

**Reply by anne8rete** [reply]
Thanks, seems though that fixed() function doesn't have the constant parameter: 
Error in fixed(0.3, constant = T) : unused argument (constant = T)

**Reply by Siel** [reply]
It's weird, for me it does, and I just checked the code and the parameter is there

**Reply by anne8rete** [reply]
I've reloaded Pmetrics, at it doesn't give that error now, but still getting into the same fortan error. Meaning the model loads, but doesn't start the iterations. 

dyld[97292]: dyld cache '/System/Library/dyld/dyld_shared_cache_arm64e' not loaded: syscall to map cache into shared region failed
dyld[97292]: Library not loaded: /usr/lib/libSystem.B.dylib

...

rm: CHMAX*.*: No such file or directory
Loading required package: Pmetrics

Attaching package: ‘Pmetrics’

The following object is masked from ‘package:base’:

    range

Error in if (success) { : argument is of length zero

**Reply by Siel** [reply]
I’ve seen that error before sometimes when I install Pmetrics and the package is still loaded. Please try restarting the R session and reinstalling Pmetrics again

**Reply by anne8rete** [reply]
I've just done it, still not running - it does run with old Pmetrics, when I fix the parameter with "!"
Getting the same fortran error.

**Reply by Siel** [reply]
@anne8rete would you help me with something? creade your model object and then use $write() over it, that function should create a file called genmodel.txt in your working directory is that model different? Thanks!

**Reply by anne8rete** [reply]
Yes, so the run starts and already creates a genmodel.txt file and to me it looks the same: 

#pri
V, 0.000000, 200.000000
K12, 0.000000, 10.000000
K21, 0.000000, 10.000000
Vmax, 0.000000, 100.000000
Km, 0.000000, 80.000000
K34, 0.000000, 10.000000
K43, 0.000000, 10.000000
SCLM, 0.000000, 100.000000
ratio, 0.300000!

#sec
V2 = ratio*V

#dif
XP(1)=RATEIV(1) - (K12 + Vmax/(Km*V+X(1))) *X(1) + K21*X(2)
XP(2)=K12*X(1) - K21*X(2)
XP(3)=(Vmax/(Km*V+X(1)))*X(1) - K43*X(4) - K34*X(3) - (SCLM/V2)*X(3)
XP(4)=K34*X(3) - K43*X(4)

#out
Y(1)=X(1)/V
Y(2)=X(3)/V2

#err
G=5.000000
0.010000,0.100000,0.000000,0.000000
0.010000,0.100000,0.000000,0.000000
